### PR TITLE
Basic implementation of cutscene

### PR DESCRIPTION
**Pull request by LoicNassif**
Still need to implement a bunch of features, but wanted to get some feedback on the basic structure I've implemented so far.

### CODE REVIEW

File: src/game/cutscene.cpp [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-
+	m_position = 0;
+	m_curr_img = &m_sprites[m_position];
+	m_state = state;
 }
 
 Cutscene::~Cutscene() {
-
+	changeState(m_state);
+	delete[] &m_sprites;
```
Comment (darren-moore) [inline src/game/cutscene.cpp]: `m_sprites` is not an array, but an `std::vector`. There is no need to delete these.

File: src/game/cutscene.cpp [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 
 void Cutscene::handleEvent(const SDL_Event& e) {
-
+	if (e.type == SDL_SCANCODE_RIGHT && m_position < size(m_sprites)-1) m_curr_img = &m_sprites[m_position++];
+	else if (e.type == SDL_SCANCODE_LEFT && m_position > 0) m_curr_img = &m_sprites[m_position--];
 }
 
 void Cutscene::update(int delta) {
-
+	render(); // Render the current image
```
Comment (darren-moore) [inline src/game/cutscene.cpp]: I think `render` is called from the engine. No need to call it here.

**Comment by darren-moore** [comment]
You'll need to consider how sprites are added to the slideshow. (you might have to change your `vector` to hold `Sprite*`)

File: src/game/cutscene.cpp (hunk 2 above)
Comment (LoicNassif) [reply src/game/cutscene.cpp]: Done

File: src/game/cutscene.cpp (hunk 1 above)
Comment (LoicNassif) [reply src/game/cutscene.cpp]: Done

**Comment by LoicNassif** [comment]
Wait, did I break something?

**Pull request merged**